### PR TITLE
Feature/pbr next: Transmission (GSVN-15)

### PR DESCRIPTION
**Pull request by UX3D-kanzler**

### CODE REVIEW

File: src/Renderer/renderer.js [hunk 1]
```diff
@@ -25,6 +25,13 @@ class gltfRenderer
         this.webGl = new gltfWebGl();
         this.webGl.context = context;
 
+        // create render target for non transmission materials
+        this.opaqueRenderTexture = 0;
+        this.opaqueFramebuffer = 0;
+        this.opaqueDepthTexture = 0;
+        this.opaqueFramebufferWidth = 1024;
+        this.opaqueFramebufferHeight = 1024;
```
Comment (UX3D-becher) [inline src/Renderer/renderer.js:33]: maybe we should just call them after transmission, opaque could be confused with the traditional opaque / alpha blending approach

File: src/Renderer/renderer.js [hunk 2]
```diff
@@ -296,6 +388,22 @@ class gltfRenderer
...
+            this.webGl.context.bindTexture(this.webGl.context.TEXTURE_2D, this.opaqueRenderTexture);
+            this.webGl.context.uniform1i(this.shader.getUniformLocation("u_TransmissionFramebufferSampler"), textureCount);
+            textureCount++;
+
+            this.webGl.context.uniform2i(this.shader.getUniformLocation("u_ScreenSize"), this.currentWidth, this.currentHeight);
+            this.webGl.context.uniform2i(this.shader.getUniformLocation("u_TransmissionFramebufferSize"), this.opaqueFramebufferWidth, this.opaqueFramebufferHeight);
+
+            //this.webGl.context.activeTexture(WebGL2RenderingContext.TEXTURE0 + textureCount);
+            //this.webGl.context.bindTexture(this.webGl.context.TEXTURE_2D, this.opaqueDepthTexture);
+            //this.webGl.context.uniform1i(this.shader.getUniformLocation("u_TransmissionDepthSampler"), textureCount);
```
Comment (UX3D-becher) [inline src/Renderer/renderer.js:403]: the need for depth would arise when we use a more sophisticated algorithm?

File: src/Renderer/shaders/textures.glsl [hunk 3]
```diff
@@ -70,6 +70,9 @@ uniform mat3 u_SheenRoughnessUVTransform;
 uniform sampler2D u_TransmissionSampler;
 uniform int u_TransmissionUVSet;
 uniform mat3 u_TransmissionUVTransform;
+uniform sampler2D u_TransmissionFramebufferSampler;
+uniform ivec2 u_TransmissionFramebufferSize;
+//uniform sampler2D u_TransmissionDepthSampler;
```
Comment (UX3D-becher) [inline src/Renderer/shaders/textures.glsl:75]: only keep if it is a helpful hint for future implementation

**Review by UX3D-becher — APPROVED** [review]

File: src/Renderer/renderer.js (hunk 1 above)
Comment (UX3D-kanzler) [reply src/Renderer/renderer.js:33]: This framebuffer for now contains all opaque elements. "Opaque" in this context excludes transmissive and alpha-blended objects. So I think, we can keep this name

**Pull request merged**